plaidml/op/lib/ops.cc

```diff
@@ -2905,7 +2905,7 @@ Value squeeze(const Value& value) {
     }
   }
   std::vector<Value> O_dims_values;
-  for (const auto dim : O_dims) {
+  for (const auto& dim : O_dims) {
```

Comment:
This should not be by reference, `Value` is a smart-pointer but it has value-like semantics.

Reply:
Sorry, I misread this code. I thought we were looping over `Value`, but we are looping over `TensorDim`, sorry. Avoiding the use of `auto` here would be useful in this case for making it easier to review.

Reply:
The type is TensorDim. Here is the error from the compiler if it is not ra ref...
plaidml/op/lib/ops.cc:2908:19: error: loop variable 'dim' of type 'const plaidml::edsl::TensorDim' creates a copy from type 'const plaidml::edsl::TensorDim' [-Werror,-Wrange-loop-analysis]
  for (const auto dim : O_dims) {
                  ^
plaidml/op/lib/ops.cc:2908:8: note: use reference type 'const plaidml::edsl::TensorDim &' to prevent copying
  for (const auto dim : O_dims) {
       ^~~~~~~~~~~~~~~~
                  &